lib/Sema/SemaExpr.cpp

```diff
@@ -11935,7 +11935,20 @@ ExprResult Sema::ActOnCountBoundsExpr(SourceLocation BoundsKWLoc,
                                       CountBoundsExpr::Kind Kind,
                                       Expr *CountExpr,
                                       SourceLocation RParenLoc) {
-  return new (Context) CountBoundsExpr(Kind, CountExpr, BoundsKWLoc, RParenLoc);
+  ExprResult Result = UsualUnaryConversions(CountExpr);
+  if (Result.isInvalid())
+    return ExprError();
+  Expr *PromotedCountExpr = Result.get();
+  QualType ResultType = PromotedCountExpr->getType();
+  if (!ResultType->isIntegerType()) {
+    std::string CountName = (Kind == CountBoundsExpr::Kind::ElementCount)
+      ? "count" : "byte_count";
```

Comment:
I agree - we should avoid embedding direct literals into the code.   Unfortunately, these are context-sensitive keywords defined in a header file for the parser.  That header file isn't pulled into Sema.h and it'd probably be a layering violation to do so.    I've created two error messages instead.  clang has no localization of error messages, but if it did, this would make it easier :).